File: python/app/models/video.py
#app/models/video.py
from app.database import db
from datetime import datetime

class Video(db.Model):
    __tablename__ = 'videos'
    
    video_id = db.Column(db.String(20), unique=True, nullable=False, primary_key=True)
    title = db.Column(db.String(255))
    duration = db.Column(db.Integer)  # Duration in seconds
    thumbnail = db.Column(db.VARCHAR(255))
    created_at = db.Column(db.DateTime, default=datetime.now)
    updated_at = db.Column(db.DateTime, default=datetime.now, onupdate=datetime.now)
    clips = db.relationship('Clip', back_populates='video', cascade='all, delete-orphan')

File: src/types/Video.ts
/**
 * Represents a video
 * @interface Video
 * @property {string} videoId - Video ID
 * @property {number} clips - Number of clips
 * @property {number} duration - Duration of the video in seconds
 * @property {string} title - Title of the video
 * @property {string} thumbnail - Thumbnail URL of the video
 * @property {Date} created_at - Date the video was created
 * @property {Date} updated_at - Date the video was last updated
 */
export interface Video {
    videoId: string;
    clips: number;
    duration: number;
    title: string;
    thumbnail: string;
    created_at: Date;
    updated_at: Date;
}